fe/src/main/java/org/apache/doris/rewrite/FoldConstantsRule.java

```diff
@@ -48,6 +50,12 @@
 
     @Override
     public Expr apply(Expr expr, Analyzer analyzer) throws AnalysisException {
+        // evaluate case when expr
+        if (expr instanceof CaseExpr && expr.getChild(0) instanceof BoolLiteral) {
```

Comment:
This is not right, the CaseExpr has multi child
1. optional case expr
2. optional list<when expr, then expr>
3. optional else expr.

The following SQL will return wrong result after applying your rewrite rule:
```
select case when false then 2 when true then 3 else 0 end;
```
Expect: 3, return 1.